examples/tutorials/testing.md

```diff
@@ -0,0 +1,729 @@
+- Remote files (those not starting with `file:`)
+
+This ensures your coverage reports focus on your application code rather than
+test files or external dependencies.
+
+### Coverage Configuration
+
+You can exclude files from coverage reports by using the `--exclude` flag:
+
+```bash
+deno coverage --exclude="test_,vendor/,_build/,node_modules/" coverage_dir
+```
+
+### Integrating with CI
+
+For continuous integration environments, you might want to enforce a minimum
+coverage threshold:
+
+```yaml
+# In your GitHub Actions workflow
+- name: Run tests with coverage
+  run: deno test --coverage=coverage_dir
+
+- name: Check coverage meets threshold
+  run: |
+    COVERAGE=$(deno coverage coverage_dir | grep "total:" | grep -o '[0-9]\+\.[0-9]\+')
+    if (( $(echo "$COVERAGE < 80" | bc -l) )); then
+      echo "Test coverage is below 80%: $COVERAGE%"
+      exit 1
+    fi
+```
+
+When working on your test coverage, remember to set realistic goals, aim for
+meaningful coverage with high quality tests over 100% coverage.
+
+## Comparison with other testing frameworks
+
+If you're coming from other JavaScript testing frameworks, here's how Deno's
+testing capabilities compare:
+
+| Feature          | Deno             | Jest                   | Mocha                      | Jasmine               |
+| ---------------- | ---------------- | ---------------------- | -------------------------- | --------------------- |
+| Setup            | Built-in         | Requires installation  | Requires installation      | Requires installation |
+| Syntax           | `Deno.test()`    | `test()`, `describe()` | `it()`, `describe()`       | `it()`, `describe()`  |
+| Assertions       | From std library | Built-in expect        | Requires assertion library | Built-in expect       |
+| Mocking          | From std library | Built-in jest.mock()   | Requires sinon or similar  | Built-in spies        |
+| Async support    | Native           | Needs special handling | Supports promises          | Supports promises     |
+| File watching    | `--watch` flag   | watch mode             | Requires nodemon           | Requires extra tools  |
+| Code coverage    | Built-in         | Built-in               | Requires istanbul          | Requires istanbul     |
+| Parallel testing | Supported        | Supported              | Limited support            | Limited support       |
+
+### Testing Style Comparison
+
+**Deno:**
+
+```ts
+import { assertEquals } from "jsr:@std/assert";
+
+Deno.test("add function", () => {
+  assertEquals(1 + 2, 3);
+});
+```
+
+**Jest:**
+
+```ts
+test("add function", () => {
+  expect(1 + 2).toBe(3);
+});
+```
+
+**Mocha:**
+
+```ts
+import { assert } from "chai";
+
+describe("math", () => {
+  it("should add numbers", () => {
+    assert.equal(1 + 2, 3);
+  });
+});
+```
+
+**Jasmine:**
+
+```ts
+describe("math", () => {
+  it("should add numbers", () => {
+    expect(1 + 2).toBe(3);
+  });
+});
+```
+
```

Comment:
Nit: wondering if we should mention our bdd API testing here from `@std/testing`

Reply:
I was gonna touch on bdd in another tut <3